src/node_file.cc

```diff
@@ -854,6 +872,43 @@ void FromNamespacedPath(std::string* path) {
 #endif
 }
 
+static inline int GetValidMode(Environment* env,
+                               Local<Value> mode_v,
+                               std::string_view type) {
+  if (!mode_v->IsInt32() && !mode_v->IsNullOrUndefined()) {
+    THROW_ERR_INVALID_ARG_TYPE(env, "mode must be int32 or null/undefined");
+    return -1;
+  }
```

Comment:
Can probably be a CHECK because callers already validate the type, I believe?

Reply:
This is the first time caller validates it. We remove the validation from JS to C++ with this pull request.